tests/lib/SimpleSAML/Utils/SystemTest.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+namespace SimpleSAML\Test\Utils;
+
+use \SimpleSAML_Configuration as Configuration;
+use \SimpleSAML\Utils\System;
+
+use \org\bovigo\vfs\vfsStream;
+
+/**
+ * Tests for SimpleSAML\Utils\System.
+ */
+class SystemTest extends \PHPUnit_Framework_TestCase
+{
+    const ROOTDIRNAME = 'testdir';
+
+    public function setUp()
+    {
+        $this->root = vfsStream::setup(
+            self::ROOTDIRNAME,
+            null,
+            array(
+                'tempdir' => array(),
+            )
+        );
+        $this->root_directory = vfsStream::url(self::ROOTDIRNAME);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::getOS
+     * @test
+     */
+    public function testGetOSBasic()
+    {
+        $res = System::getOS();
+
+        $this->assertInternalType("int", $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::resolvePath
+     * @test
+     */
+    public function testResolvePathRemoveTrailingSlashes()
+    {
+        $base = "/base////";
+        $path = "test";
+
+        $res = System::resolvePath($path, $base);
+        $expected = "/base/test";
+
+        $this->assertEquals($expected, $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::resolvePath
+     * @test
+     */
+    public function testResolvePathPreferAbsolutePathToBase()
+    {
+        $base = "/base/";
+        $path = "/test";
+
+        $res = System::resolvePath($path, $base);
+        $expected = "/test";
+
+        $this->assertEquals($expected, $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::resolvePath
+     * @test
+     */
+    public function testResolvePathCurDirPath()
+    {
+        $base = "/base/";
+        $path = "/test/.";
+
+        $res = System::resolvePath($path, $base);
+        $expected = "/test";
+
+        $this->assertEquals($expected, $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::resolvePath
+     * @test
+     */
+    public function testResolvePathParentPath()
+    {
+        $base = "/base/";
+        $path = "/test/child/..";
+
+        $res = System::resolvePath($path, $base);
+        $expected = "/test";
+
+        $this->assertEquals($expected, $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::writeFile
+     * @test
+     */
+    public function testWriteFileInvalidArguments()
+    {
+        $this->setExpectedException('\InvalidArgumentException');
+        System::writeFile(null, null, null);
+    }
+
+    /**
+     * @requires PHP 5.4.0
```

Comment:
This is a limitation of `vfsstream`: https://github.com/mikey179/vfsStream/issues/56

```
touch(), chown(), chgrp() and chmod() only work with PHP >= 5.4 because support of stream wrappers for those functions was first added with PHP 5.4, see issue #56. With PHP 5.3 an error with the message "Unable to create file vfs://... because No such file or directory" will be raised
```

Other than this, `vfsstream` is a nice library. I've used it for a number of projects and it's definitely helpful when mocking out the filesystem :)